src/e2e-test/java/io/cdap/plugin/bqexecute/stepsdesign/BQExecuteConnector.java

```diff
@@ -0,0 +1,160 @@
+  }
+
+  @Then("Toggle BigQueryExecute property Use Cache")
+  public void toggleBigQueryExecutePropertyUseCache() {
+    BQExecuteActions.clickUseCache();
+  }
+
+  @Then("Select BigQueryExecute property mode {string}")
+  public void selectBigQueryExecutePropertyMode(String value) {
+    BQExecuteActions.selectMode(value);
+  }
+
+  @Then("Enter GCS source property path field {string}")
+  public void enterGCSSourcePropertyPathField(String pathField) {
+    CdfGcsActions.enterPathField(PluginPropertyUtils.pluginProp(pathField));
+  }
+
+  @Then("Close the BigQueryExecute Properties")
+  public void closeTheBigQueryExecuteProperties() {
+    BQExecuteActions.closeButton();
+  }
+
+  @Then("Enter BigQueryExecute source property table name")
+  public void enterBigQueryExecuteSourcePropertyTableName() {
+    BQExecuteActions.enterBQExeTableName(TestSetupHooks.bqTargetTable);
+  }
+
+  @Then("Enter BigQueryExecute property dataset {string}")
+  public void enterBigQueryExecutePropertyDataset(String dataset) {
+    BQExecuteActions.enterDatasetName(PluginPropertyUtils.pluginProp(dataset));
+  }
+
+  @Then("Enter BigQueryExecute property datasetProjectId {string}")
+  public void enterBigQueryExecutePropertyDatasetProjectId(String projectId) throws IOException {
+    BQExecuteActions.enterProjectId(PluginPropertyUtils.pluginProp(projectId));
+  }
+
+  @Then("Enter BigQueryExecute {string} as macro argument {string}")
+  public void enterBigQueryExecuteAsMacroArgument(String pluginProperty, String macroArgument) {
+    enterPropertyAsMacroArgument(pluginProperty, macroArgument);
+  }
+
+  @Then("Enter runtime argument value for BigQuery Executer table name key {string}")
+  public void enterRuntimeArgumentValueForBigQueryExecuterTableNameKey(String runtimeArgumentKey) {
+    ElementHelper.sendKeys(CdfStudioLocators.runtimeArgsValue(runtimeArgumentKey), TestSetupHooks.bqTargetTable);
+  }
+
+  @Then("Validate records transferred to target BigQuery Table is equal to number of records fetched from " +
+    "SQL query {string}")
+  public void validateRecordsTransferredToTargetBigQueryTableIsEqualToNumberOfRecordsFetchedFromSQLQuery(String query)
+    throws IOException, InterruptedException {
+    String projectId = (PluginPropertyUtils.pluginProp("projectId"));
+    String datasetName = (PluginPropertyUtils.pluginProp("dataset"));
+    int countRecordsTarget = BigQueryClient.countBqQuery(TestSetupHooks.bqTargetTable);
+    String selectQuery = (PluginPropertyUtils.pluginProp(query).replace
+        ("DATASET", PluginPropertyUtils.pluginProp("dataset"))
+      .replace("PROJECT_NAME", PluginPropertyUtils.pluginProp
+        ("projectId"))
+      .replace("TABLENAME", TestSetupHooks.bqSourceTable));
+    Optional<String> result = BigQueryClient.getSoleQueryResult(selectQuery);
+    int count = result.map(Integer::parseInt).orElse(0);
+    BeforeActions.scenario.write("Number of records transferred with respect to :" + count);
```

Comment:
-> BeforeActions.scenario.write("Number of records fetched with sql query :" + count);

Reply:
@sawantpritam -Updated